src/search.cpp

```diff
@@ -77,51 +78,57 @@ InternalDataBase::InternalDataBase(const std::vector<Archive>& archives, bool ve
+                    // So we need a language, let's use the one of the zim.
+                    // If zimfile has no language metadata, we can't do lot more here :/
+                    try {
+                        language = archive.getMetadata("Language");
+                    } catch(...) {}
                 }
-            }
-            auto stopwords = database.get_metadata("stopwords");
-            if ( !stopwords.empty() ){
-                std::string stopWord;
-                std::istringstream file(stopwords);
-                Xapian::SimpleStopper* stopper = new Xapian::SimpleStopper();
-                while (std::getline(file, stopWord, '\n')) {
-                    stopper->add(stopWord);
+                if (!language.empty()) {
+                    icu::Locale languageLocale(language.c_str());
+                    /* Configuring language base steemming */
+                    try {
+                        m_stemmer = Xapian::Stem(languageLocale.getLanguage());
+                        m_queryParser.set_stemmer(m_stemmer);
+                        m_queryParser.set_stemming_strategy(Xapian::QueryParser::STEM_ALL);
+                    } catch (...) {
+                        std::cout << "No stemming for language '" << languageLocale.getLanguage() << "'" << std::endl;
+                    }
+                }
+                auto stopwords = database.get_metadata("stopwords");
+                if ( !stopwords.empty() ){
+                    std::string stopWord;
+                    std::istringstream file(stopwords);
+                    Xapian::SimpleStopper* stopper = new Xapian::SimpleStopper();
+                    while (std::getline(file, stopWord, '\n')) {
+                        stopper->add(stopWord);
+                    }
+                    stopper->release();
+                    m_queryParser.set_stopper(stopper);
+                }
+            } else {
+                std::map<std::string, int> valuesmap = read_valuesmap(database.get_metadata("valuesmap"));
+                if (m_valuesmap != valuesmap ) {
+                    // [TODO] Ignore the database, raise a error ?
                 }
-                stopper->release();
-                m_queryParser.set_stopper(stopper);
-            }
-        } else {
-            std::map<std::string, int> valuesmap = read_valuesmap(database.get_metadata("valuesmap"));
-            if (m_valuesmap != valuesmap ) {
-                // [TODO] Ignore the database, raise a error ?
             }
+            m_xapianDatabases.push_back(database);
+            m_database.add_database(database);
+            m_archives.push_back(archive);
+            first = false;
+        } catch( Xapian::DatabaseError& e ) {
+            // [TODO] Ignore the database or raise a error ?
+            // As we already ignore the database if `getDbFromAccessInfo` "detects" a DatabaseError,
+            // we also ignore here.
         }
```

Comment:
> Otherwise, we'd better first come up with such a rule.

May be time to do so.

I tend to say that we should suppres the "Format" error (corrupted archives...) and we should bubble the error (potentially using only one exception class (ZimFileFormatError)).
(But I'm open to your opinion)

But in this Pr, I haven't try to enforce this rule and try to keep the same "API" as before : when errors were suppressed I continue suppress it and when exceptions were bubble up, I have just throw a ZimFileFormatError instead.